modules/ztp-image-based-upgrade-prep.adoc

```diff
@@ -0,0 +1,208 @@
+
+// are there other prereqs?
+
+.Procedure
+
+This example procedure demonstrates how to back up and upgrade a cluster with applications on a persistent volume.
+
+[NOTE]
+====
+The target cluster does not need to be detached from the hub cluster.
+====
+
+. (Optional) Create your OADP `Backup` and `Restore` CRs.
+
+.. Define the restore order for the OADP Operator in the `Restore` CR by using the `lca.openshift.io/apply-wave` field:
++
+[source,yaml]
+----
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: acm-klusterlet
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "1"
+spec:
+  backupName:
+    acm-klusterlet
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: example-app
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "3"
+spec:
+  backupName:
+    example-app
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: localvolume
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "2"
+spec:
+  backupName:
+    localvolume
+----
+
+.. Create a `kustomization.yaml` that will append the information to a new `ConfigMap` CR:
++
+[source,yaml]
+----
+configMapGenerator:
+- name: oadp-cm-example
+  namespace: openshift-adp
+  files:
+  - backup-acm-klusterlet.yaml
+  - backup-localvolume.yaml
+  - backup-app.yaml
+  - restore-acm-klusterlet.yaml
+  - restore-localvolumes.yaml
+  - restore-app.yaml
+generatorOptions:
+  disableNameSuffixHash: true <1>
+----
+<1> Disables the hash generation at the end of the `ConfigMap` filename which allows the `ConfigMap` CR to be overwritten when a new `ConfigMap` CR is generated with the same name.
+
+.. Create the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ kustomize build ./ -o oadp-cm.yaml
+----
++
+.Example output
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: oadp-cm-example
+  namespace: openshift-adp
+[...]
+----
+
+.. Apply the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ oc apply -f oadp-cm.yaml
+----
+
+. Edit the `ImageBasedUpgrade` CR:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: example-upgrade
+spec:
+  stage: Idle
+  seedImageRef:
+    version: 4.14.7 <1>
+    image: <seed_container_image> <2>
+    pullSecretRef: <seed_pull_secret> <3>
+  additionalImages:
+    name: ""
+    namespace: ""
+  autoRollbackOnFailure: {} <4>
+#    disabledForPostRebootConfig: "true" <5>
+#    disabledForUpgradeCompletion: "true" <6>
+#    disabledInitMonitor: "true" <7>
+#    initMonitorTimeoutSeconds: 1800 <8>
+#  extraManifests: <9>
+#  - name: sno-extra-manifests
+#    namespace: openshift-lifecycle-agent
+#  oadpContent: <10>
+#  - name: oadp-cm-example
+#    namespace: openshift-adp
+----
+<1> Specify the platform version on the seed image.
```

Comment:
We need to expand on it. 
The version must match the seedimage version or the upgrade will fail complaining about mismatch